lib/vega_lite/wx_viewer.ex

```diff
@@ -0,0 +1,47 @@
+if Code.ensure_compiled(:wx) == {:module, :wx} do
+  defmodule VegaLite.WxViewer do
+    @moduledoc false
+
+    @behaviour :wx_object
+
+    @title "VegaLite graphic"
+    @size {1000, 800}
+
+    @doc """
+    Creates a graphical window displaying the given HTML.
+
+    Returns `{:ok, pid}` on success, where `pid` identifies
+    the process managing the window widget.
+    """
+    @spec start(binary()) :: {:ok, pid()} | :error
+    def start(html) do
+      case :wx_object.start(__MODULE__, [html: html], []) do
+        {:error, _reason} -> :error
+        {:wx_ref, _id, _obj, pid} -> {:ok, pid}
+      end
+    end
+
+    @impl true
+    def init(opts) do
+      html = opts[:html]
+
+      wx = :wx.new()
+
+      frame = :wxFrame.new(wx, -1, @title, size: @size)
+      :wxFrame.connect(frame, :close_window)
+
+      web_view = :wxWebView.new(frame, -1)
+      :wxWebView.setPage(web_view, html, "")
+
```

Comment:
Hmm the warnings indicate that `:wxWebView` may not be present even if `:wx` is, so we should extend the condition then.